File: station-registry/src/main/java/md/vnastasi/cloud/ApplicationProperties.java
package md.vnastasi.cloud;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

@ConfigurationProperties(prefix = "application")
@ConstructorBinding
public record ApplicationProperties(
    NsApiProperties nsApi
) {

    public record NsApiProperties(
        String baseUrl,
        PublicTravelInfoProperties publicTravelInfo
    ) {

        public record PublicTravelInfoProperties(
            String basePath,
            ApiKeyProperties apiKey,
            String stationsPath
        ) {

            public record ApiKeyProperties(
                String header,
                String value
            ) {
            }
        }
    }
}

File: disruptions/src/main/kotlin/md/vnastasi/cloud/ApplicationProperties.kt
package md.vnastasi.cloud

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "application")
data class ApplicationProperties(
    val nsApi: NsApiProperties
)

data class NsApiProperties(
    val publicTravelInfo: PublicTravelInfoProperties,
    val baseUrl: String
)

data class PublicTravelInfoProperties(
    val apiKey: ApiKeyProperties,
    val basePath: String,
    val disruptionsPath: String
)

data class ApiKeyProperties(
    val header: String,
    val value: String
)
